cross-validation/logistic-regression/script.whizzml

```diff
@@ -0,0 +1,404 @@
+;; This code will eventually be defined as a library.
+
+(define MODEL_OPTIONS ["balance_objective"
+                       "missing_splits"
+                       "pruning"
+                       "weight_field"
+                       "objective_weights"
+                       "node_threshold"])
+(define ENSEMBLE_OPTIONS (concat MODEL_OPTIONS
+                                 ["sample_rate"
+                                  "replacement"
+                                  "randomize"
+                                  "number_of_models"
+                                  "seed"]))
+(define LOGISTIC_OPTIONS ["balance_fields"
+                          "bias"
+                          "c"
+                          "missing_numerics"
+                          "default_numeric_value"
+                          "eps"
+                          "field_codings"
+                          "normalize"
+                          "regularization"
+                          "seed"])
+(define EVALUATION_OPTIONS ["sample_rate"
+                            "out_of_bag"
+                            "range"
+                            "replacement"
+                            "ordering"
+                            "seed"
+                            "missing_strategy"
+                            "combiner"])
+;; cross-validation
+;;
+;; creates k-fold cross-validation for a dataset
+;; Inputs:
+;;   dataset-id: (string) Dataset ID
+;;   k-folds: (integer) Number of folds
+;;   model-options: (map) options to use in model/ensemble
+;;   evaluation-options: (map) options to use in evaluation creation
+;;
+;; Output: (map) Average of evaluations results
+;;
+;; Raises:
+;;  101: The dataset-id argument is not a string
+;;  102: The dataset-id is not a valid dataset ID
+;;  103: The k-folds argument is not an integer
+;;  104: The k-folds argument is not >= 2
+;;  106: The objective field ID is not in the selectable IDs list
+;;
+(define (cross-validation dataset-id
+                          k-folds
+                          objective-id
+                          model-options
+                          evaluation-options)
+  (resource-ID-ok? dataset-id "dataset")
```

Comment:
if this is just checking and raising, a better name would be something like `check-resource-id`